aesara/tensor/subtensor_opt.py

```diff
@@ -1666,3 +1668,100 @@ def local_subtensor_SpecifyShape_lift(fgraph, node):
         new_shape_arg = as_tensor(shape_arg[indices], dtype=np.int64, ndim=1)
 
     return [specify_shape(new_obj_arg, new_shape_arg)]
+
+
+def equal_idxs(idx1, idx2):
+    if isinstance(idx1, slice) and isinstance(idx2, slice):
+        return (
+            equal_computations([idx1.start], [idx2.start])
+            and equal_computations([idx1.stop], [idx2.stop])
+            and equal_computations([idx1.step], [idx2.step])
+        )
+    else:
+        return equal_computations([idx1], [idx2])
+
+
+@register_useless
+@register_canonicalize
+@register_specialize
+@local_optimizer([Join])
+def local_join_subtensors(fgraph, node):
+    """Simplify contiguous subtensors inside a join
```

Comment:
```suggestion
    r"""Simplify contiguous :class:`Subtensor`\s inside a :class:`Join`.
```

Keep in mind that we need to be more consistent about the format and formatting of our docstrings.